lib/nerves_runtime/init.ex

```diff
@@ -98,11 +98,19 @@ defmodule Nerves.Runtime.Init do
 
   defp unmount_if_error(s), do: s
 
-  defp format_if_unmounted(%{mounted: :unmounted} = s) do
+  defp format_if_unmounted(%{mounted: :unmounted, fstype: "f2fs"} = s) do
     "Formatting application partition. If this hangs, it could be waiting on the urandom pool to be initialized"
     |> Logger.warn()
 
-    System.cmd("mkfs.#{s.fstype}", ["-U", @app_partition_uuid, "-F", "#{s.devpath}"])
+    System.cmd("mkfs.f2fs", ["#{s.devpath}"])
```

Comment:
Would you mind one more pass on this to refactor the `System.cmd` line into a helper function? Something like:

```elixir
defp mkfs("f2fs", devpath) do
  System.cmd("mkfs.f2fs", [devpath])
end
defp mkfs(_fstype, devpath) do
  System.cmd("mkfs.#{fstype}", ["-U", @app_partition_uuid, "-F", devpath])
end
```
This gist is to remove the duplicate log message and to simplify the function head for `format_if_unmounted` to only concern itself with the mount status.

Reply:
Thanks for the feedback. I agree the smaller more focused functions are more readable.